src/rust/src/expr/string.rs

```diff
@@ -0,0 +1,396 @@
+use crate::{prelude::*, PlRExpr, RPolarsErr};
+use savvy::{savvy, NumericScalar, Result, Sexp};
+
+#[savvy]
+impl PlRExpr {
+    fn str_len_bytes(&self) -> Result<Self> {
+        Ok(self.inner.clone().str().len_bytes().into())
+    }
+
+    fn str_len_chars(&self) -> Result<Self> {
+        Ok(self.inner.clone().str().len_chars().into())
+    }
+
+    fn str_join(&self, delimiter: &str, ignore_nulls: bool) -> Result<Self> {
+        Ok(self
+            .inner
+            .clone()
+            .str()
+            .join(delimiter, ignore_nulls)
+            .into())
+    }
+
+    fn str_to_uppercase(&self) -> Result<Self> {
+        Ok(self.inner.clone().str().to_uppercase().into())
+    }
+
+    fn str_to_lowercase(&self) -> Result<Self> {
+        Ok(self.inner.clone().str().to_lowercase().into())
+    }
+
+    // fn str_to_titlecase(&self) -> Result<Self> {
+    //     f_str_to_titlecase(self)
+    // }
+
+    fn str_strip_chars(&self, matches: &PlRExpr) -> Result<Self> {
+        Ok(self
+            .inner
+            .clone()
+            .str()
+            .strip_chars(matches.inner.clone())
+            .into())
+    }
+
+    fn str_strip_chars_end(&self, matches: &PlRExpr) -> Result<Self> {
+        Ok(self
+            .inner
+            .clone()
+            .str()
+            .strip_chars_end(matches.inner.clone())
+            .into())
+    }
+
+    fn str_strip_chars_start(&self, matches: &PlRExpr) -> Result<Self> {
+        Ok(self
+            .inner
+            .clone()
+            .str()
+            .strip_chars_start(matches.inner.clone())
+            .into())
+    }
+
+    fn str_zfill(&self, alignment: &PlRExpr) -> Result<Self> {
+        Ok(self
+            .inner
+            .clone()
+            .str()
+            .zfill(alignment.inner.clone())
+            .into())
+    }
+
+    fn str_pad_end(&self, width: NumericScalar, fillchar: &str) -> Result<Self> {
+        let width = <Wrap<usize>>::try_from(width)?.0;
```

Comment:
Since there is no equivalent in R, I think it is better to define and use something like `impl TryFrom<&str> for Wrap<char>` as like this:
https://github.com/eitsupi/neo-r-polars/blob/c26e768f5b2436b0d4ffb8a6c465672ad6a9102b/src/rust/src/conversion/mod.rs#L261

Reply:
Done, I think the wording of the message might be improved if you have some suggestions. All tests pass now

Reply:
Looks great!

Notifying user which argument is the problem is what is missing in the current error handling.
These will need to be improved later, but for now it seems perfect.